src/Common/src/Microsoft/Win32/SafeHandles/SafeCreateHandle.OSX.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Win32.SafeHandles
+{
+    /// <summary>
+    /// This class is a wrapper around the Create pattern in OS X where
+    /// if a Create* function is called, the caler must also CFRelease 
+    /// on the same pointer in order to correctly free the memory.
+    /// </summary>
+    [System.Security.SecurityCritical]
+    internal sealed partial class SafeCreateHandle : SafeHandle
+    {
+        internal SafeCreateHandle(IntPtr ptr) : base(IntPtr.Zero, true)
+        {
+            this.SetHandle(ptr);
+        }
+
+        [System.Security.SecurityCritical]
+        protected override bool ReleaseHandle()
+        {
+            if (IsInvalid == false)
```

Comment:
While it shouldn't hurt anything to have it, we shouldn't need this check inside of Release, and pretty much none of the SafeHandle implementations in the core libraries do such a check.